src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1844,17 +1845,8 @@ private static EmitResult ToEmitResultAndFree(DiagnosticBag diagnostics, bool su
                     {
                         Debug.Assert(Options.StrongNameProvider != null);
 
-                        // Targeted try-catch for errors during CreateInputStream as found in TFS 1140649
-                        // TODO: Put this wrapping in PeWriter to catch all potential PE writing exceptions
-                        try
-                        {
-                            signingInputStream = Options.StrongNameProvider.CreateInputStream();
-                            retStream = signingInputStream;
-                        }
-                        catch (Exception e)
-                        {
-                            throw new Cci.PeWritingException(e);
-                        }
+                        signingInputStream = Options.StrongNameProvider.CreateInputStream();
+                        retStream = signingInputStream;
```

Comment:
Because this will get caught by the catch in PEWriter/PeWriter.cs.  Notice that this code is inside of a lambda.  The lambda gets executed later; note the comment above the try.